python-client/giskard/datasets/base/__init__.py

```diff
@@ -328,38 +323,41 @@ def _infer_column_types(self, cat_columns: List[str]):
         Returns:
             dict: A dictionary that maps column names to their inferred types, one of 'text', 'numeric', or 'category'.
         """
-        column_types = {}
-        nuniques = self.df.nunique()
-        df_columns = list(self.df.columns)
+        if not column_types:
+            column_types = {}
```

Comment:
Either this is useless, or the type hint in the function signature is not correct.